src/saveload/afterload.cpp

```diff
@@ -3080,6 +3080,11 @@ bool AfterLoadGame()
 		}
 	}
 
+	if (IsSavegameVersionBefore(SLV_TOWN_CARGOGEN)) {
+		/* Ensure the original cargo generation mode is used */
+		_settings_game.economy.town_cargogen_mode = TCGM_ORIGINAL;
+	}
+
```

Comment:
Is this necessary? These forced-settings for old versions used to be a firm exception, not standard.

Reply:
The default is the new mode, and old saves are converted to keep using the old mode. The intention is to not break profitability of existing networks - if your passenger network assumes 20k passengers per station per month, suddenly getting only 5k passengers is going to be a major hit.

Reply:
my opinion is that we shouldn't keep the original mode, but if we do keep it, it kinda makes sense to set old savegames to that mode.